javascript/new-src/src/js/app.js

```diff
@@ -14,665 +14,698 @@
+      $ola.get.ItemList().then(function(data) {
+        $scope.Universes = data.universes;
       });
-     },
-     AddUniverse: function (data) {
-      return $http({
-       method: 'POST',
-       url: '/new_universe',
-       data: postEncode(data),
-       headers: {
-        'Content-Type': 'application/x-www-form-urlencoded'
-       }
-      }).then(function (response) {
-       return response.data;
+      $ola.get.ServerInfo().then(function(data) {
+        $scope.Info = data;
       });
-     },
-     Dmx: function (universe, dmx) {
-      var data = {
-       u: universe,
-       d: dmxConvert(dmx)
+      $scope.Shutdown = function() {
+        $ola.action.Shutdown().then();
       };
-      return $http({
-       method: 'POST',
-       url: '/set_dmx',
-       data: postEncode(data),
-       headers: {
-        'Content-Type': 'application/x-www-form-urlencoded'
-       }
-      }).then(function (response) {
-       return response.data;
-      });
-     }
-    },
-    action: {
-     Shutdown: function () {
-      return $http.get('/quit').then(function (response) {
-       return response.data;
-      });
-     },
-     Reload: function () {
-      return $http.get('/reload').then(function (response) {
-       return response.data;
+      $scope.goUniverse = function(id) {
+        $location.path('/universe/' + id);
+      };
+    }])
+  .controller('addUniverseCtrl', ['$scope', '$ola', '$window', '$location',
+    function($scope, $ola, $window, $location) {
+      'use strict';
+      $scope.Ports = {};
+      $scope.addPorts = [];
+      $scope.Universes = [];
+      $scope.Class = '';
+      $scope.Data = {
+        id: 0,
+        name: '',
+        add_ports: ''
+      };
+      $ola.get.ItemList().then(function(data) {
+        for (var u in data.universes) {
+          if (data.universes.hasOwnProperty(u)) {
+            if ($scope.Data.id === parseInt(data.universes[u].id, 10)) {
+              $scope.Data.id++;
+            }
+            $scope.Universes.push(parseInt(data.universes[u].id, 10));
+          }
+        }
       });
-     },
-     ReloadPids: function () {
-      return $http.get('/reload_pids').then(function (response) {
-       return response.data;
+      $scope.Submit = function() {
+        if (typeof $scope.Data.id === 'number' &&
+          $scope.Data.add_ports !== '' &&
+          $scope.Universes.indexOf($scope.Data.id) === -1) {
+          if ($scope.Data.name === undefined || $scope.Data.name === '') {
+            $scope.Data.name = 'Universe ' + $scope.Data.id;
+          }
+          $ola.post.AddUniverse($scope.Data);
+          $location.path('/universe/' + $scope.Data.id);
+        } else if ($scope.Universes.indexOf($scope.Data.id) !== -1) {
+          $ola.error.modal('Universe Id already exists.');
+        } else if ($scope.Data.add_ports === undefined ||
+          $scope.Data.add_ports === '') {
+          $ola.error.modal('There are no ports selected for the universe.' +
```

Comment:
Nit, but can the space from the line below fit on this line, which will look nicer?